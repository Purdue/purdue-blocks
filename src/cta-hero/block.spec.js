jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock,
} from '@wordpress/e2e-test-utils';

import {
    clickElementByText,
    blockStartup,
    updateRangeInput,
    clickCheckbox,
    clickRadio,
    openSidebarPanelWithTitle
} from '../test-helpers'

const block = {blockTitle: 'CTA Hero', blockName: 'purdue-blocks/cta-hero'}

describe( 'ðŸ”¬ CTA Hero Block', () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock( block.blockTitle )


        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    describe('ðŸ”¬ Block Editor Fields', () => {
        test('ðŸ”Ž Page Title Field', async () =>{
            await blockStartup(block)

            const typeString = "CTA Hero page title test."
            
            await page.focus('input[placeholder="Page Title..."]')
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()
            expect( await getEditedPostContent() ).toMatchSnapshot()
            expect( editedContent.includes(`"pageTitle":"${typeString}"`)).toBe(true)
        })

        test('ðŸ”Ž Media Library button', async () => {
            await blockStartup(block)

            // open media library
            await clickElementByText('button', 'Open Media Library')

            // tests that media library does open
            expect(await page.$(`div.media-frame-title`)).not.toBeNull()
        })

        describe('ðŸ”¬ CTA Button Settings Fields', () => {
            test('ðŸ”Ž CTA text field', async () => {
                await blockStartup(block)

                const typeString = "CTA Hero button text test."
    
                await page.focus('input[placeholder="CTA Text..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"ctaText":"${typeString}"`)).toBe(true)
            })
            test('ðŸ”Ž CTA url field', async () => {
                await blockStartup(block)

                const typeString = "https://www.purdue.edu"
    
                await page.focus('input[placeholder="CTA URL..."]')
                await page.keyboard.type(typeString, {delay: 10})
    
                const editedContent = await getEditedPostContent()
                expect( await getEditedPostContent() ).toMatchSnapshot()
                expect( editedContent.includes(`"ctaUrl":"${typeString}"`)).toBe(true)
            })
        })
    })

    describe('ðŸ”¬ Side Panel Settings', () => {
        
        test( 'ðŸ”Ž Image alt text field.', async () => {
            await blockStartup(block)

            const typeString = "Image alt text."
            
            // focus the text box then type into it with the virtual keyboard
            await clickElementByText('label', 'Hero Image Alt Text')
            await page.keyboard.type(typeString, {delay: 10})

            const editedContent = await getEditedPostContent()
            expect( editedContent).toMatchSnapshot()
            expect( editedContent.includes(`aria-label="${typeString}"`)).toBe(true)
            expect( editedContent.includes(`"altText":"${typeString}"`)).toBe(true)
        })
    })
})