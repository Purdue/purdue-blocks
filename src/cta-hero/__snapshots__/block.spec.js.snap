// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ”¬ CTA Hero Block ðŸ”Ž Block should be available. 1`] = `
"<!-- wp:purdue-blocks/cta-hero -->
<div class=\\"pu-cta-hero\\"><div class=\\"hero is-large\\"><div class=\\"hero-body\\"><div class=\\"background-image\\" aria-label=\\"\\"><style>
              .background-image {background-image: url();}
              @media (max-width: 767px) {
                .background-image {background-image: url();}
              }
            </style></div><div class=\\"container\\"><div class=\\"content\\"><h1></h1><a href=\\"\\" class=\\"pu-cta-hero__button\\"></a></div></div></div></div></div>
<!-- /wp:purdue-blocks/cta-hero -->"
`;

exports[`ðŸ”¬ CTA Hero Block ðŸ”¬ Block Editor Fields ðŸ”Ž Page Title Field 1`] = `
"<!-- wp:purdue-blocks/cta-hero {\\"pageTitle\\":\\"CTA Hero page title test.\\"} -->
<div class=\\"pu-cta-hero\\"><div class=\\"hero is-large\\"><div class=\\"hero-body\\"><div class=\\"background-image\\" aria-label=\\"\\"><style>
              .background-image {background-image: url();}
              @media (max-width: 767px) {
                .background-image {background-image: url();}
              }
            </style></div><div class=\\"container\\"><div class=\\"content\\"><h1>CTA Hero page title test.</h1><a href=\\"\\" class=\\"pu-cta-hero__button\\"></a></div></div></div></div></div>
<!-- /wp:purdue-blocks/cta-hero -->"
`;

exports[`ðŸ”¬ CTA Hero Block ðŸ”¬ Block Editor Fields ðŸ”¬ CTA Button Settings Fields ðŸ”Ž CTA text field 1`] = `
"<!-- wp:purdue-blocks/cta-hero {\\"ctaText\\":\\"CTA Hero button text test.\\"} -->
<div class=\\"pu-cta-hero\\"><div class=\\"hero is-large\\"><div class=\\"hero-body\\"><div class=\\"background-image\\" aria-label=\\"\\"><style>
              .background-image {background-image: url();}
              @media (max-width: 767px) {
                .background-image {background-image: url();}
              }
            </style></div><div class=\\"container\\"><div class=\\"content\\"><h1></h1><a href=\\"\\" class=\\"pu-cta-hero__button\\">CTA Hero button text test.</a></div></div></div></div></div>
<!-- /wp:purdue-blocks/cta-hero -->"
`;

exports[`ðŸ”¬ CTA Hero Block ðŸ”¬ Block Editor Fields ðŸ”¬ CTA Button Settings Fields ðŸ”Ž CTA url field 1`] = `
"<!-- wp:purdue-blocks/cta-hero {\\"ctaUrl\\":\\"https://www.purdue.edu\\"} -->
<div class=\\"pu-cta-hero\\"><div class=\\"hero is-large\\"><div class=\\"hero-body\\"><div class=\\"background-image\\" aria-label=\\"\\"><style>
              .background-image {background-image: url();}
              @media (max-width: 767px) {
                .background-image {background-image: url();}
              }
            </style></div><div class=\\"container\\"><div class=\\"content\\"><h1></h1><a href=\\"https://www.purdue.edu\\" class=\\"pu-cta-hero__button\\"></a></div></div></div></div></div>
<!-- /wp:purdue-blocks/cta-hero -->"
`;

exports[`ðŸ”¬ CTA Hero Block ðŸ”¬ Side Panel Settings ðŸ”Ž Image alt text field. 1`] = `
"<!-- wp:purdue-blocks/cta-hero {\\"altText\\":\\"Image alt text.\\"} -->
<div class=\\"pu-cta-hero\\"><div class=\\"hero is-large\\"><div class=\\"hero-body\\"><div class=\\"background-image\\" aria-label=\\"Image alt text.\\"><style>
              .background-image {background-image: url();}
              @media (max-width: 767px) {
                .background-image {background-image: url();}
              }
            </style></div><div class=\\"container\\"><div class=\\"content\\"><h1></h1><a href=\\"\\" class=\\"pu-cta-hero__button\\"></a></div></div></div></div></div>
<!-- /wp:purdue-blocks/cta-hero -->"
`;
